charts/tfy-karpenter-config/templates/karpenter-inferentia-provisioner.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if .Values.karpenter.inferentiaProvisionerSpec.enabled }}
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: inferentia-provisioner
+  namespace: karpenter
+spec:
+  weight: 10
+  {{- if .Values.karpenter.inferentiaProvisionerSpec.consolidation.enabled }}
+  consolidation:
+    enabled: true
+  {{- else }}
+  ttlSecondsAfterEmpty: {{ .Values.karpenter.inferentiaProvisionerSpec.ttlSecondsAfterEmpty }}
+  {{- end }}
+  requirements:
+    - key: karpenter.sh/capacity-type
+      operator: In
+      values:
+      {{- range .Values.karpenter.inferentiaProvisionerSpec.capacityTypes }}
+        - {{ . | quote }}
+      {{- end }}
+    - key: topology.kubernetes.io/zone
+      operator: In
+      values:
+      {{- range .Values.karpenter.inferentiaProvisionerSpec.zones  }}
+        - {{ . | quote }}
+      {{- end }}
+    - key: kubernetes.io/arch
+      operator: In
+      values:
+        - amd64
+    - key: karpenter.k8s.aws/instance-family
+      operator: In
+      values:
+      {{- range .Values.karpenter.inferentiaProvisionerSpec.instanceFamilies  }}
+        - {{ . | quote }}
+      {{- end }}
+    - key: karpenter.k8s.aws/instance-size
+      operator: NotIn
+      values:
+      {{- range .Values.karpenter.inferentiaProvisionerSpec.instanceSizes.notAllowed  }}
+        - {{ . | quote }}
+      {{- end }}
+  taints:
+    - key: "aws.amazon.com/neuron"
+      effect: "NoSchedule"
```

Comment:
Are we making sure these taints are handled by the tolerations from the platform.

Reply:
We will ensure it.